bumble/profiles/hap.py

```diff
@@ -0,0 +1,590 @@
+
+                    @watcher.on(connection, 'disconnection')
+                    def on_disconnection() -> None:
+                        self.currently_connected_clients.remove(connection)
+
+                self.currently_connected_clients.add(connection)
+                if (
+                    self.preset_changed_operations_history_per_device.get(
+                        connection.device.public_address
+                    )
+                    == None
+                ):
+                    self.preset_changed_operations_history_per_device[
+                        connection.device.public_address
+                    ] = []
+                    return
+                # Send all the PresetChangedOperation that occur when not connected
+                await self._preset_changed_operation(connection)
+                # Update the active preset index if needed
+                await self.notify_active_preset_for_connection(connection)
+
+        self.hearing_aid_features_characteristic = gatt.Characteristic(
+            uuid=gatt.GATT_HEARING_AID_FEATURES_CHARACTERISTIC,
+            properties=gatt.Characteristic.Properties.READ,
+            permissions=gatt.Characteristic.Permissions.READ_REQUIRES_ENCRYPTION,
+            value=bytes(self.hearing_aid_features),
+        )
+        self.hearing_aid_preset_control_point = gatt.Characteristic(
+            uuid=gatt.GATT_HEARING_AID_PRESET_CONTROL_POINT_CHARACTERISTIC,
+            properties=(
+                gatt.Characteristic.Properties.WRITE
+                | gatt.Characteristic.Properties.INDICATE
+            ),
+            permissions=gatt.Characteristic.Permissions.WRITE_REQUIRES_ENCRYPTION,
+            value=gatt.CharacteristicValue(
+                write=self._on_write_hearing_aid_preset_control_point
+            ),
+        )
+        self.active_preset_index_characteristic = gatt.Characteristic(
+            uuid=gatt.GATT_ACTIVE_PRESET_INDEX_CHARACTERISTIC,
+            properties=(
+                gatt.Characteristic.Properties.READ
+                | gatt.Characteristic.Properties.NOTIFY
+            ),
+            permissions=gatt.Characteristic.Permissions.READ_REQUIRES_ENCRYPTION,
+            value=bytes([self.active_preset_index]),
+        )
+
+        super().__init__(
+            [
+                self.hearing_aid_features_characteristic,
+                self.hearing_aid_preset_control_point,
+                self.active_preset_index_characteristic,
+            ]
+        )
+
+    # TODO this need to be triggered when device is unbonded
+    def on_forget(self, addr: Address):
+        self.preset_changed_operations_history_per_device.pop(addr)
+
+    def _on_write_hearing_aid_preset_control_point(
+        self, connection: Optional[Connection], value: bytes
+    ):
+        opcode = HearingAidPresetControlPointOpcode(value[0])
+        handler = getattr(self, '_on_' + opcode.name.lower())
+        handler(connection, value)
```

Comment:
Since your handlers are asynchrounous, you need to use them with abort_on, otherwise they won't work.
```suggestion
        connection.abort_on('disconnection', handler(connection, value))
```